storage/app/src/main/java/com/google/firebase/referencecode/storage/KotlinStorageActivity.kt

```diff
@@ -0,0 +1,405 @@
+package com.google.firebase.referencecode.storage
+
+import android.graphics.Bitmap
+import android.graphics.drawable.BitmapDrawable
+import android.net.Uri
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.widget.ImageView
+import com.google.android.gms.tasks.Continuation
+import com.google.android.gms.tasks.Task
+import com.google.firebase.referencecode.storage.interfaces.StorageActivityInterface
+import com.google.firebase.storage.FirebaseStorage
+import com.google.firebase.storage.StorageMetadata
+import com.google.firebase.storage.StorageReference
+import com.google.firebase.storage.UploadTask
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.FileInputStream
+
+class KotlinStorageActivity : AppCompatActivity(), StorageActivityInterface {
+    // [START storage_field_declaration]
+    // [END storage_field_declaration]
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_storage)
+
+        // [START storage_field_initialization]
+        val storage = FirebaseStorage.getInstance()
+        // [END storage_field_initialization]
+
+        includesForCreateReference()
+    }
+
+    override fun includesForCreateReference() {
+        val storage = FirebaseStorage.getInstance()
+
+        // ## Create a Reference
+
+        // [START create_storage_reference]
+        // Create a storage reference from our app
+        var storageRef = storage.reference
+        // [END create_storage_reference]
+
+        // [START create_child_reference]
+        // Create a child reference
+        // imagesRef now points to "images"
+        var imagesRef: StorageReference? = storageRef.child("images")
+
+        // Child references can also take paths
+        // spaceRef now points to "images/space.jpg
+        // imagesRef still points to "images"
+        var spaceRef = storageRef.child("images/space.jpg")
+        // [END create_child_reference]
+
+        // ## Navigate with References
+
+        // [START navigate_references]
+        // getParent allows us to move our reference to a parent node
```

Comment:
We have comments here referencing methods like `getRoot` and `getParent` but since we're using the property annotation (`.root` and `.parent`) we should change the comments.